source/postprocess/particle_distribution_statistics.cc

```diff
@@ -0,0 +1,360 @@
+                            "The granularity parameter determines how many discrete inputs exist for "
+                            "the probability density function generated by the kernel density estimator. "
+                            "The domain of the function is multidimensional so the granularity value determines "
+                            "the range of inputs in each dimension. For example, a granularity value of 2 "
+                            "results in a PDF which is defined for the inputs 0-1 in each of its dimensions. ");
+          prm.declare_entry("Use KDE per particle","false",
+                            Patterns::Bool(),
+                            "If `KDE_per_particle` is true, the point-density function is defined at the position "
+                            "of every particle in the cell. If it is false, the point-density-function is defined "
+                            "on a regular grid throughout the cell."
+                           );
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+    }
+
+
+
+    template <int dim>
+    void
+    ParticleDistributionStatistics<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Particle distribution statistics");
+        {
+          KDE_per_particle = prm.get_bool("Use KDE per particle");
+          granularity = prm.get_integer("KDE granularity");
+          bandwidth = prm.get_double("Kernel bandwidth");
+          std::string kernel_function_string = prm.get("Kernel function");
+
+          if (kernel_function_string =="Triangular")
+            {
+              kernel_function = KernelFunctions::TRIANGULAR;
+            }
+          else if (kernel_function_string =="Gaussian")
+            {
+              kernel_function = KernelFunctions::GAUSSIAN;
+            }
+          else if (kernel_function_string == "CutoffFunctionW1")
+            {
+              kernel_function = KernelFunctions::CUTOFF_FUNCTION_W1_DEALII;
+            }
+          else
+            {
+              kernel_function = KernelFunctions::UNIFORM;
+            }
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(ParticleDistributionStatistics,
+                                  "Particle Distribution Statistics",
+                                  "A postprocessor that computes some statistics about "
+                                  "the particle distribution within grid cells. In particular "
+                                  "it calculates a point-density function per cell and derives "
+                                  "the standard deviation of that function. It also reports the "
+                                  "maximum and minimum values of the point-density function.")
```

Comment:
I assume you are taking the stddev/min/max of the values *across all cells*? Perhaps make that clearer: You get one value per cell, and you're then taking statistics over all cells.